### PR TITLE
fix(DTFS2-7604): residual comments for PR #3897

### PR DESCRIPTION
**Pull request by abhi-markan**
## Introduction :pencil2:
Few comments were not addressed in the PR #3897 

## Resolution :heavy_check_mark:
* Added fallback assignment using logical OR operator for `_id` (Mongo deal ID) property access, pertinent to diverse deal types.

## Miscellaneous ➕ 
* Removed `jest.clearAllMocks()`;
* Updated JS Docs.
* Reworded Jest `describe` label.
* Updated no Mongo deal ID mock.

### CODE REVIEW

**Review by ttbarnes — DISMISSED** [review]
Great work!

**Review by ttbarnes — DISMISSED** [review]

File: trade-finance-manager-api/src/v1/helpers/amendment.helpers.js [hunk 1]
```diff
@@ -331,19 +331,21 @@ const firstTaskEmailConfirmation = async (facilityId, amendmentId, auditDetails)
...
- * @param {Object} auditDetails - The audit details for logging purposes.
+ * @param {import("@ukef/dtfs2-common").AuditDetails}
  * @returns {Promise<void>} - A promise that resolves when the email has been sent and the flag has been updated.
  */
 const sendFirstTaskEmail = async (deal, auditDetails) => {
   try {
     const { amendment, dealSnapshot, facilityId, amendmentId } = deal;
     const { tasks } = amendment;
-    const { _id: dealId } = deal;
+    const { _id } = deal;
```
Comment (ChristianMcc) [inline trade-finance-manager-api/src/v1/helpers/amendment.helpers.js:341]: you could keep this re-assignment to dealId to make it clear, maybe even dealMongoId, but ye good catch the assignment was wrong way round

`const { dealMongoId: _id } = deal;`

Comment (ChristianMcc) [reply trade-finance-manager-api/src/v1/helpers/amendment.helpers.js:341]: probably not necessary just thought that's probably where the error is more than anything else! we can keep the logic on line 346 of using the dealSnapshot and just getting whichever one we can if we need

File: trade-finance-manager-api/src/v1/helpers/amendment.helpers.js [hunk 2]
```diff
@@ -331,19 +331,21 @@ const firstTaskEmailConfirmation = async (facilityId, amendmentId, auditDetails)
...
     const { companyName } = exporter;
 
-    // dealId in snapshot for gef and details for bss
-    const ukefDealId = dealSnapshot.ukefDealId || dealSnapshot?.details?.ukefDealId;
+    // BSS/EWCS or GEF Mongo deal ID
+    const dealId = dealSnapshot?._id || _id;
+    // BSS/EWCS or GEF UKEF deal ID
+    const ukefDealId = dealSnapshot?.details?.ukefDealId || dealSnapshot.ukefDealId;
     const firstTask = getFirstTask(tasks);
     const templateId = EMAIL_TEMPLATE_IDS.TASK_READY_TO_START;
```
Comment (ChristianMcc) [inline trade-finance-manager-api/src/v1/helpers/amendment.helpers.js:351]: it's not part of this PR but just fwiw something like:

`Missing required parameters to send email for ${dealId}` or `Missing required arguments provided for ${dealId}` would be a slightly clearer error message on line 353

**Review by ChristianMcc — APPROVED** [review]
LGTM, this code is absolutely fine so approving but left a couple of comments if you want to change anything before merging or just in retrospect

File: trade-finance-manager-ui/server/controllers/case/index.test.js [hunk 3]
```diff
@@ -181,7 +181,7 @@ describe('controllers - case', () => {
       });
     });
 
-    describe('should catch any thrown error', () => {
+    describe('when an error is thrown', () => {
```
Comment (AlexBramhill) [inline trade-finance-manager-ui/server/controllers/case/index.test.js:184]: Thank you!

**Review by AlexBramhill — DISMISSED** [review]
lgtm, thanks!

File: trade-finance-manager-api/src/v1/helpers/amendment.helpers.js (hunk 2 above)
Comment (abhi-markan) [reply trade-finance-manager-api/src/v1/helpers/amendment.helpers.js:351]: Good shout I have reworded it.

File: trade-finance-manager-api/src/v1/helpers/amendment.helpers.js (hunk 1 above)
Comment (abhi-markan) [reply trade-finance-manager-api/src/v1/helpers/amendment.helpers.js:341]: Good shout, have moved the logic to LOC:346 should have thought about it.

File: trade-finance-manager-api/src/v1/__mocks__/amendmentVariables.js [hunk 4]
```diff
@@ -226,10 +226,17 @@ const dealWithNoCompanyName = {
     bankInternalRefName: 'UKEF',
   },
 };
-
-const dealWithNoId = {
+const dealWithNoMongoId = {
```
Comment (abhi-markan) [inline trade-finance-manager-api/src/v1/__mocks__/amendmentVariables.js:229]: To be explicit on which Id as `dealWithNoId` can be confused for either MongoId or UKEF Deal ID.

**Review by ChristianMcc — DISMISSED** [review]
LGTM! :)
